client/src/components/AddCard/AddCardForm/AddCardForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useEffect } from 'react';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import { Formik, FormikHelpers } from 'formik';
+import * as Yup from 'yup';
+import Typography from '@material-ui/core/Typography';
+import useStyles from './useStyles';
+import { useAuth } from '../../../context/useAuthContext';
+import { CircularProgress } from '@material-ui/core';
+import { addCard } from '../../../helpers/APICalls/addCard';
+import { getAllPaymentMethods } from '../../../helpers/APICalls/getAllPaymentMethods';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import { useHistory } from 'react-router-dom';
+import CreditCard from './CreditCard/CreditCard';
+
+interface Props {
+  handleSubmit: (
+    {
+      paymentMethod,
+    }: {
+      paymentMethod: string;
+    },
+    {
+      setStatus,
+      setSubmitting,
+    }: FormikHelpers<{
+      paymentMethod: string;
+    }>,
+  ) => void;
+}
+
+export default function AddCard({ handleSubmit }: Props): JSX.Element {
+  const [paymentMethods, setPaymentMethods] = useState<any>([]);
+  const { loggedInUser, loggedInUserProfile } = useAuth();
+  const { updateSnackBarMessage } = useSnackBar();
+  const history = useHistory();
+
+  useEffect(() => {
+    if (loggedInUser) {
+      getAllPaymentMethods({ userId: loggedInUser.id }).then((data) => {
+        if (data.error) {
+          updateSnackBarMessage(data.error.message);
+        } else if (data.allPaymentMethods) {
+          setPaymentMethods(data.allPaymentMethods);
+        } else {
+          updateSnackBarMessage('An unexpected error occurred. Please try again');
+        }
+      });
+    }
+  }, [loggedInUser, updateSnackBarMessage]);
+
+  const handleAddCard = () => {
+    if (loggedInUser) {
+      addCard({ userId: loggedInUser.id }).then((data) => {
+        if (data.error) {
+          updateSnackBarMessage(data.error.message);
+        } else if (data.customer) {
+          history.push({
+            pathname: '/add-card-info',
+            state: {
+              clientSecret: data.intent.client_secret,
+            },
+          });
+        } else {
+          updateSnackBarMessage('An unexpected error occurred. Please try again');
+        }
+      });
+    }
+  };
+  const classes = useStyles();
```

Comment:
Nitpick: You can move this line of code above (line 38)

Reply:
Done.